crates/uv-dispatch/src/lib.rs

```diff
@@ -316,7 +317,17 @@ impl<'a> BuildContext for BuildDispatch<'a> {
     ) -> Result<SourceBuild> {
         match self.no_build {
             NoBuild::All => debug_assert!(
-                matches!(build_kind, BuildKind::Editable),
+                match self.no_binary {
+                    // Allow `all` to be overridden by specific binary exclusions
+                    NoBinary::Packages(packages) => {
+                        if let Some(dist) = dist {
+                            packages.contains(dist.name())
+                        } else {
+                            false
+                        }
+                    }
+                    _ => false,
+                } || matches!(build_kind, BuildKind::Editable),
```

Comment:
nit: Can move the part after the or to the front? I got confused since when match has closures

Reply:
Should be resolved in #4284